### ISSUE TITLE
orchestrations parallel_runners module not recognizing profile

### ISSUE THREAD

**Issue by alan-cugler** [saltstack/salt#51263]
### Description of Issue/Question
Hi, I don't understand the format I need in the orchestration state to get parallel_runners to recognize the profile file. I can get these runners working individually in a regular orchestration state.
### Setup
The provider configuration
```YAML
#/etc/salt/cloud.providers.d/digitalocean-provider.conf
digitalocean:
  driver: digitalocean
  personal_access_token: [ Redacted ]
  ssh_key_file: [ Redacted ]
  ssh_key_names: [ Redacted ]
```
The profile configuration
```YAML
#/etc/salt/cloud.profiles.d/digitalocean-profile.conf
do-ubuntu-1604-2gb-nyc1:
  provider: digitalocean
  image: 16.04.5 x64
  size: 2gb
  location: nyc1
  private_networking: True
  backups_enabled: False
  ipv6: False
  create_dns_record: False
  ...
```
```YAML
#srv/salt/orch/test.sls
startup_vms:                                                                  
  salt.parallel_runners:                                                      
    - runners:                                                                
        masters_1_install:                                                    
          - name: cloud.profile                                               
          - prof: do-ubuntu-1604-2gb-nyc1                                     
          - provider: digitalocean                                            
          - instances:                                                        
            - master11                                                        
          - kwarg:                                                            
              master:                                                         
                user: root                                                    
              script: bootstrap-salt                                          
              script_args: -N -M                                              
        masters_23_install:                                                   
          - name: cloud.profile                                               
          - prof: do-ubuntu-1604-2gb-nyc1                                     
          - provider: digitalocean                                            
          - instances:                                                        
            - master22                                                        
            - master33                                                        
          - kwarg:                                                            
              master:                                                         
                user: root                                                    
              script: bootstrap-salt                                          
              script_args: -N -S -M                                           
        minions_1_to_3:                                                       
          - name: cloud.profile                                               
          - prof: do-ubuntu-1604-2gb-nyc1                                     
          - provider: digitalocean                                            
          - instances:                                                        
            - minion11                                                        
            - minion22                                                        
            - minion33
```
### Steps to Reproduce Issue
```YAML
$ salt-run state.orch orch.test
master1:
----------
          ID: startup_vms
    Function: salt.parallel_runners
      Result: True
     Comment: All runner functions executed successfully.
     Started: 11:20:39.894409
    Duration: 726.656 ms
     Changes:   
              ----------
              ret:
                  ----------
                  masters_1_install:
                      ----------
                      Error:
                          A profile (or prof) must be defined
                  masters_23_install:
                      ----------
                      Error:
                          A profile (or prof) must be defined
                  minions_1_to_3:
                      ----------
                      Error:
                          A profile (or prof) must be defined

Summary for master1
------------
Succeeded: 1 (changed=1)
Failed:    0
------------
Total states run:     1
Total run time: 726.656 ms
```
### Versions Report
```
Salt Version:
           Salt: 2018.3.3
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.6.1
      docker-py: Not Installed
          gitdb: 2.0.3
      gitpython: 2.1.8
          ioflo: Not Installed
         Jinja2: 2.10
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.7
   msgpack-pure: Not Installed
 msgpack-python: 0.5.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.15rc1 (default, Nov 12 2018, 14:31:15)
   python-gnupg: 0.4.1
         PyYAML: 3.13
          PyZMQ: 16.0.2
           RAET: Not Installed
          smmap: 2.0.3
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.2.5
 
System Versions:
           dist: Ubuntu 18.04 bionic
         locale: UTF-8
        machine: x86_64
        release: 4.15.0-43-generic
         system: Linux
        version: Ubuntu 18.04 bionic
```
### Resources
* [Orchestration Runner](https://docs.saltstack.com/en/latest/topics/orchestrate/orchestrate_runner.html)
* [Parallel Runner](https://docs.saltstack.com/en/latest/ref/states/all/salt.states.saltmod.html#salt.states.saltmod.parallel_runners)
* [Cloud Profile Module](https://docs.saltstack.com/en/latest/ref/states/all/salt.states.cloud.html#salt.states.cloud.absent)

**Comment by Ch3LL** [comment]
you need to pass the instance, prof, etc into the kwarg:

```
startup_vms:
  salt.parallel_runners:
    - runners:
        masters_1_install:
          - name: cloud.profile
          - kwarg:                                                            
              prof: do-ubuntu-1604-2gb-nyc1
              provider: digitalocean
              instances:
                - master11
              master:
                user: root 
              script: bootstrap-salt
              script_args: -N -M
```

**Comment by alan-cugler** [comment]
So I ran the kwarg change for my whole script above. The first three virtual machines spun up all at the same time, one from each runner. After the first three spun up in parallel then the rest spun up sequentially. 

Is there no way to tell orchestration to evaluate a runner, and if multiple virtual machines are listed for that runner to then spin all of those up in parallel? If not I'd like to make that a feature request.

**Comment by Ch3LL** [comment]
so you have more than 3 VMs setup in the same `salt.parallel_runners` state and they aren't running parallel? is that correct? I would assume they would run parallel

**Comment by alan-cugler** [comment]
I have three runners indented below `salt.parallel_runners`, I had expected orchestration to evaluate all three runners and determine there were 6 virtual machines to be created and spin them all up at once. Instead it spun up 3 virtual machines in parallel (one from each runner) and then sequentially spun up the rest of the virtual machines.
```YAML
startup_vms:
  salt.parallel_runners:
    - runners:
        masters_1_install:
          - name: cloud.profile
          - kwarg:
              prof: do-ubuntu-1604-2gb-nyc1
              provider: digitalocean
              instances:
              - master11
              master:
                user: root
              script: bootstrap-salt
              script_args: -N -M
        masters_23_install:
          - name: cloud.profile
          - kwarg:
              prof: do-ubuntu-1604-2gb-nyc1
              provider: digitalocean
              instances:
              - master22
              - master33
              master:
                user: root
              script: bootstrap-salt
              script_args: -N -S -M
        minions_1_to_3:
          - name: cloud.profile
          - kwarg:
              prof: do-ubuntu-1604-2gb-nyc1
              provider: digitalocean
              instances:
              - minion11
              - minion22
              - minion33
```

**Comment by alan-cugler** [comment]
I talked to Gary Richmond about it and he had me add `parallel: True` to each runner. This gives me the 6 virtual machines spinning up at once. This is what I wanted, though it's a bit buggy.

Successful orch parallel.runner state:
```YAML
startup_vms:                                                                      
  salt.parallel_runners:                                                          
    - runners:                                                                    
        masters_1_install:                                                        
          - name: cloud.profile                                                   
          - kwarg:                                                                
              prof: do-ubuntu-1604-2gb-nyc1                                       
              provider: digitalocean                                              
              instances:                                                          
              - master11                                                          
              master:                                                             
                user: root                                                        
              script: bootstrap-salt                                              
              script_args: -N -M                                                  
        masters_23_install:                                                       
          - name: cloud.profile                                                   
          - kwarg:                                                                
              prof: do-ubuntu-1604-2gb-nyc1                                       
              parallel: True                                                      
              provider: digitalocean                                              
              instances:                                                          
              - master22                                                          
              - master33                                                          
              master:                                                             
                user: root                                                        
              script: bootstrap-salt                                              
              script_args: -N -S -M                                               
        minions_1_to_3:                                                           
          - name: cloud.profile                                                   
          - kwarg:                                                                
              prof: do-ubuntu-1604-2gb-nyc1                                       
              parallel: True                                                      
              provider: digitalocean                                              
              instances:                                                          
              - minion11                                                          
              - minion22                                                          
              - minion33
```

Since this gives me what I was looking for I will close this. However, if the I cant figure out the bug for some failed cluster parallel spin ups, I will make a new issue for it.
